perfscale_regression_ci/scripts/encryption/toggle_encryption.sh

```diff
@@ -0,0 +1,148 @@
+    else
+        log "Still waiting for: encryption.apiserver.operator.openshift.io/migrated-resources in ${secret_file} secret."
+        continue_wait=true
+    fi
+
+    if [[ ${secret_timestamp} != "" ]]; then
+        log "Founded: encryption.apiserver.operator.openshift.io/migrated-timestamp in ${secret_file} secret."
+    else
+        log "Still waiting for: encryption.apiserver.operator.openshift.io/migrated-timestamp in ${secret_file} secret."
+        continue_wait=true
+    fi
+
+    if [[ ${secret_mode} == "${mode}" ]]; then
+        log "Founded: encryption.apiserver.operator.openshift.io/mode in ${kb_secrets_file} secret."
+    else
+        log "Still waiting for: encryption.apiserver.operator.openshift.io/mode in ${kb_secrets_file} secret."
+        continue_wait=true
+    fi
+}
+
+function wait_until_encryption_is_ready {
+    start_time=$(date +%s)
+    while ((($(date +%s) - start_time) < timeout)); do
+        continue_wait=false
+
+        echo -e "Not ready yet. Time from beginning: $(($(date +%s) - start_time)) seconds"
+        echo -e "Retrying in one minute"
+        sleep 60
+
+        # Getting secrets
+        log "Getting ${oc_secrets_name} secret"
+        oc get secret "${oc_secrets_name}" -o yaml -n openshift-config-managed >"${OC_SECRET}"
+        log "Getting ${kb_secrets_name} secret"
+        oc get secret "${kb_secrets_name}" -o yaml -n openshift-config-managed >"${KB_SECRET}"
+        log "Getting ${oauth_secrets_name} secret"
+        oc get secret "${oauth_secrets_name}" -o yaml -n openshift-config-managed >"${OAUTH_SECRET}"
+
+        # Checking encryption-key-openshift-apiserver secret
+        log "------- Checking ${oc_secrets_name} secret"
+        check_secret "${OC_SECRET}" "${OC_MIGRATED_RESOURCES}" "${OC_MIGRATED_RESOURCES}"
+
+        # Checking encryption-key-openshift-kube-apiserver secret
+        log "------- Checking ${kb_secrets_name} secret"
+        check_secret "${KB_SECRET}" "${KB_MIGRATED_RESOURCES_A}" "${KB_MIGRATED_RESOURCES_B}"
+
+        # Checking encryption-key-openshift-oauth-apiserver secret
+        log "------- Checking ${oauth_secrets_name} secret"
+        check_secret "${OAUTH_SECRET}" "${OAUTH_MIGRATED_RESOURCES_A}" "${OAUTH_MIGRATED_RESOURCES_B}"
+
+        if [[ ${continue_wait} == "true" ]]; then
+            continue
+        fi
+
+        log "DONE"
+        echo -e "Time from beginning: $(($(date +%s) - start_time)) seconds"
+        exit 0
+    done
+
+    log "!!!!!!!!!!! TIMEOUT !!!!!!!!!!!"
+    log "${OC_SECRET}"
+    cat "${OC_SECRET}"
+    echo ""
+    log "${KB_SECRET}"
+    cat "${KB_SECRET}"
+    echo ""
+    log "${OAUTH_SECRET}"
+    cat "${OAUTH_SECRET}"
+    exit 1
+}
+
+# START!
+if [[ $1 == "enable" ]]; then
+    enable_encryption
+elif [[ $1 == "disable" ]]; then
+    disable_encryption
+else
+    echo -e "First argument should be 'enable' to enable encryption or 'disable' to disable encryption."
+    exit 1
+fi
+
+wait_until_encryption_is_ready
```

Comment:
@skordas When disable_encryption, is there a need to run the wait_until_encryption_is_ready?

Reply:
Yes. From validation point of view enabling or disabling look the same.